src/main/java/com/commercetools/sync/products/utils/ProductVariantAttributeUpdateActionUtils.java

```diff
@@ -75,13 +78,34 @@
                         () -> SetAttributeInAllVariants.of(newProductVariantAttribute, true)) :
                     buildUpdateAction(oldProductVariantAttributeValue, newProductVariantAttributeValue,
                         () -> SetAttribute.of(variantId, newProductVariantAttribute, true));
-            } else {
-                return attributeMetaData.isSameForAll()
```

Comment:
Because before this PR, if the new attributes list was empty no update actions would have been built. Check this comment before this PR: https://github.com/commercetools/commercetools-sync-java/pull/309/files#diff-991246687782cb226352bc27cff57377L92